tracer/src/Datadog.FeatureFlags.OpenFeature/DatadogProvider.cs

```diff
@@ -0,0 +1,154 @@
+    /// <param name="cancellationToken"> Async cancelation token </param>
+    /// <returns> Returns the evaluation result </returns>
+    public override Task<ResolutionDetails<int>> ResolveIntegerValueAsync(string flagKey, int defaultValue, EvaluationContext? context = null, CancellationToken cancellationToken = default)
+    {
+        return Task.Run(
+            () =>
+            {
+                var res = Datadog.Trace.FeatureFlags.FeatureFlagsSdk.Evaluate(flagKey, Trace.FeatureFlags.ValueType.INTEGER, defaultValue, GetContext(context));
+                return GetResolutionDetails<int>(res);
+            },
+            cancellationToken);
+    }
+
+    /// <summary> Resolve flag as string </summary>
+    /// <param name="flagKey"> Requested flag </param>
+    /// <param name="defaultValue"> Default value </param>
+    /// <param name="context"> Evaluation context </param>
+    /// <param name="cancellationToken"> Async cancelation token </param>
+    /// <returns> Returns the evaluation result </returns>
+    public override Task<ResolutionDetails<string>> ResolveStringValueAsync(string flagKey, string defaultValue, EvaluationContext? context = null, CancellationToken cancellationToken = default)
+    {
+        return Task.Run(
+            () =>
+            {
+                var res = Datadog.Trace.FeatureFlags.FeatureFlagsSdk.Evaluate(flagKey, Trace.FeatureFlags.ValueType.STRING, defaultValue, GetContext(context));
+                return GetResolutionDetails<string>(res);
+            },
+            cancellationToken);
+    }
+
+    /// <summary> Resolve flag as Value </summary>
+    /// <param name="flagKey"> Requested flag </param>
+    /// <param name="defaultValue"> Default value </param>
+    /// <param name="context"> Evaluation context </param>
+    /// <param name="cancellationToken"> Async cancelation token </param>
+    /// <returns> Returns the evaluation result </returns>
+    public override Task<ResolutionDetails<Value>> ResolveStructureValueAsync(string flagKey, Value defaultValue, EvaluationContext? context = null, CancellationToken cancellationToken = default)
+    {
+        return Task.Run(
+            () =>
+            {
+                var res = Datadog.Trace.FeatureFlags.FeatureFlagsSdk.Evaluate(flagKey, Trace.FeatureFlags.ValueType.JSON, defaultValue, GetContext(context));
+                return GetResolutionDetails<Value>(res);
+            },
+            cancellationToken);
+    }
+
+    private static Datadog.Trace.FeatureFlags.IEvaluationContext? GetContext(EvaluationContext? context)
+    {
+        if (context == null) { return new Datadog.Trace.FeatureFlags.EvaluationContext(string.Empty); }
+        var values = context.AsDictionary().Select(p => new KeyValuePair<string, object?>(p.Key, ToObject(p.Value))).ToDictionary(p => p.Key, p => p.Value);
+        var res = new Datadog.Trace.FeatureFlags.EvaluationContext(context.TargetingKey!, values);
```

Comment:
`!` are a red-flag for new code. We should be handling nulls gracefully, one way or another. Either we should check for null and handle it, or it shouldn't be able to be `null`

Reply:
Fixed